linera-core/src/client/mod.rs

```diff
@@ -1396,17 +1393,14 @@ where
                 return;
             }
         }
-        for (chain_id, height) in other_sender_chains {
+        for chain_id in other_sender_chains {
             if let Err(error) = self
                 .client
                 .local_node
-                .wait_for_outgoing_messages(chain_id, height)
+                .retry_pending_cross_chain_requests(chain_id)
```

Comment:
Can you add a comment why we're retrying? It's impossible IMHO to tell why by just reading through the flows (here, the return value from `synchronize_from_validators` is used to retry pending cross chain requests.